arch/risc-v/src/common/espressif/esp_mcpwm.c

```diff
@@ -0,0 +1,682 @@
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct mcpwm_dev_common_s mcpwm_common =
+{
+  .group.group_id      = MCPWM_CAPTURE_DEFAULT_GROUP,
+  .initialized         = false,
+  .capture_initialized = false,
+  .group_prescale      = MCPWM_DEV_CLK_PRESCALE,
+};
+
+#ifdef CONFIG_ESP_MCPWM_CAPTURE
+/* Lower half methods required by the capture driver */
+
+static const struct cap_ops_s mcpwm_cap_ops =
+{
+  .start   = esp_capture_start,
+  .stop    = esp_capture_stop,
+  .getduty = esp_capture_getduty,
+  .getfreq = esp_capture_getfreq,
+};
+
+/* Data structures for the available capture channels */
+
+#ifdef CONFIG_ESP_MCPWM_CAPTURE_CH0
+static struct mcpwm_capture_event_data_s event_data_ch0;
+static struct mcpwm_cap_channel_lowerhalf_s mcpwm_cap_ch0_lowerhalf =
+{
+  .ops        = &mcpwm_cap_ops,
+  .common     = &mcpwm_common,
+  .data       = &event_data_ch0,
+  .channel_id = MCPWM_CAP_CHANNEL_0,
+  .ready      = false,
+};
+#endif
+
+#ifdef CONFIG_ESP_MCPWM_CAPTURE_CH1
+static struct mcpwm_capture_event_data_s event_data_ch1;
+static struct mcpwm_cap_channel_lowerhalf_s mcpwm_cap_ch1_lowerhalf =
+{
+  .ops        = &mcpwm_cap_ops,
+  .common     = &mcpwm_common,
+  .data       = &event_data_ch1,
+  .channel_id = MCPWM_CAP_CHANNEL_1,
+  .ready      = false,
+};
+#endif
+
+#ifdef CONFIG_ESP_MCPWM_CAPTURE_CH2
+static struct mcpwm_capture_event_data_s event_data_ch2;
+static struct mcpwm_cap_channel_lowerhalf_s mcpwm_cap_ch2_lowerhalf =
+{
+  .ops        = &mcpwm_cap_ops,
+  .common     = &mcpwm_common,
+  .data       = &event_data_ch2,
+  .channel_id = MCPWM_CAP_CHANNEL_2,
+  .ready      = false,
+};
+#endif
+#endif  /* CONFIG_ESP_MCPWM_CAPTURE */
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp_capture_start
+ *
+ * Description:
+ *   This function is a requirement of the upper-half driver. When called,
+ *   enables the capture channel, interruption routine, sets the positive
+ *   edge to trigger this interrupt and resets the frequency and duty
+ *   values. The positive edge is always the first expected.
+ *
+ * Input Parameters:
+ *   lower - Pointer to the capture channel lower-half data structure.
+ *
+ * Returned Value:
+ *   Returns OK on success.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ESP_MCPWM_CAPTURE
+static int esp_capture_start(struct cap_lowerhalf_s *lower)
+{
+  struct mcpwm_cap_channel_lowerhalf_s *priv = (
+    struct mcpwm_cap_channel_lowerhalf_s *)lower;
+  mcpwm_hal_context_t *hal = &priv->common->hal;
+
+  DEBUGASSERT(priv != NULL);
+  DEBUGASSERT(priv->common->initialized);
```

Comment:
is `DEBUGASSERT(priv->common != NULL);` needed?

Reply:
Removing.